lib/exprengine.cpp

```diff
@@ -345,7 +345,7 @@ static ExprEngine::ValuePtr getValueRangeFromValueType(const std::string &name,
 static void call(const std::vector<ExprEngine::Callback> &callbacks, const Token *tok, ExprEngine::ValuePtr value)
 {
     if (value) {
-        for (ExprEngine::Callback f : callbacks) {
+        for (const ExprEngine::Callback& f : callbacks) {
```

Comment:
Is it possible to profile such a for loop. I would be curious to know if this is better.

Reply:
I think avoiding a `std::function` copy is as superficial as the postfix/prefix in for loops (which is something Cppcheck warns about). 

Going off-topic: Regarding the latter I recently tried to optimized a function which was invoked a lot and it was using the increment with causes the copy, but changing that had zero impact on the performance. So if this even more obvious copy is not considered something to warn about I don't think the increment thing should be warned about either.

Reply:
> So if this even more obvious copy is not considered something to warn about I don't think the increment thing should be warned about either.

yes!

Reply:
I checked these with clang-tidy enabled and it reports this. So I consider keeping this change.

Reply:
Random user chiming in here, but `std::function` is allowed to internally heap allocate to store anything the function may need (e.g. If all captures in a lambda being stored into a `std::function` exceed the small internal buffer within said `std::function`) to ensure it can execute properly, so it's not necessarily superficial in general, it depends on a case-by-case basis, since copying `std::function`s unnecessarily can end up churning heap allocations in certain scenarios.

Reply:
I think Coverity does look at this - in some cases it only points out copies when a certain threshold is exceeded.